contrib/babelfishpg_tsql/src/linked_servers.c

```diff
@@ -88,6 +88,33 @@ linked_server_msg_handler(LinkedServerProcess lsproc, int error_code, int state,
 	return 0;
 }
 
+/*
+ * Handle any error encountered in TDS client library itself
+ */
+static int
+linked_server_err_handler(LinkedServerProcess lsproc, int severity, int db_error, int os_error, char *db_err_str, char *os_err_str)
+{
+	StringInfoData buf;
+
+	initStringInfo(&buf);
+
+	appendStringInfo(
+		&buf,
+		"TDS client library error: DB #: %i, DB Msg: %s, OS #: %i, OS Msg: %s, Level: %i",
+		db_error,
+		db_err_str ? db_err_str : "",
+		os_error,
+		os_err_str ? os_err_str : "",
+		severity
+	);
+
+	ereport(ERROR,
+			(errcode(ERRCODE_FDW_UNABLE_TO_CREATE_EXECUTION),
+			errmsg("%s", buf.data)));
+
+	return LS_INT_CANCEL;
```

Comment:
what is meaning of this?

Reply:
https://www.freetds.org/reference/a00554.html#ga302a4ce6559b81808e3be51615cd1b67. See INT_CANCEL

Reply:
But we have used ereport here so control will never reach back to tds_fdw, right?

Reply:
Agreed but we should return based on the behaviour intended. Theoretically, the execution should be suspended (which is what INT_CANCEL indicates). Returning 0 isn't correct because it maps to INT_EXIT which is Sybase semantics.